VoIP: Add support for announce

jaminh opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
This adds the "Announce" service to the VOIP Assist Satellite component. Currently when announce is called HA will call the phone and go into the standard assist pipeline when the call is answered. 

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

jaminh committed:
Add the assist satelite announce feature for VOIP component.

Currently when making an announcement the phone will be called and will
go into the regular assist pipeline when the call is answered.

jaminh commented:
We will need to add some way for the HassVoipDatagramProtocol to pick an appropriate protocol factory for the different use cases: HA receiving a call, HA calling out for an announcement, and HA calling out as part of the start_conversation service. 

I imagine at some point we will want to account for what happens when one of those services, announce or start_conversation, is called while there is an active call with the target VOIP device. Currently the code assumes there is not an active call and will always start by calling the phone.

balloob commented on homeassistant/components/voip/assist_satellite.py:
```diff
@@ -130,6 +147,11 @@ async def async_added_to_hass(self) -> None:
         self.voip_device.protocol = self
 
+        # Device supports announcements
+        self._attr_supported_features |= (
```
```suggestion
        self._attr_supported_features = (
```

balloob commented on homeassistant/components/voip/assist_satellite.py:
```diff
@@ -149,6 +171,83 @@ async def async_set_configuration(
+    ) -> None:
+        """Announce media on the phone. First call and wait for phone.
+
+        Before calling the phone, we should check to make sure there is not
```
I don't see this check in the code yet.

balloob commented on homeassistant/components/voip/config_flow.py:
```diff
@@ -64,13 +64,27 @@ async def async_step_init(
+                        ),
+                    ): str,
+                    vol.Required(
+                        CONF_SIP_HOST,
```
Why do we want to ask the user these values?

balloob replied:
Is this to get the IP address of the HA box? We should instead import `async_get_source_ip` from `homeassistant.components.network`

jaminh replied:
Is that guaranteed to be the external IP for the HA box? I think there would still be value in allowing the host sent in the outgoing invite message to be configurable, but using async_get_source_ip as a default value would make sense. Depending on SIP server configurations a specific user might be required in the INVITE From header, so even though that configuration isn't officially supported, it is helpful for me. If we make those optional with sane defaults is that still a workable solution?

balloob replied:
We don't want them in the main configuration flow, as users will be confused (we want to ask as little information as possible when setting up an integration). 

Can you try if it works for you with `async_get_source_ip` ?

jaminh replied:
I moved user and host to an "advanced options" page and defaulted host to the value of async_get_source_ip, I hope that is an acceptable solution?

balloob replied:
No, let's stick with just calling `async_get_source_ip` when we need to do an announcement. That method exists to get the right IP address for exactly this reason:

```python
@bind_hass
async def async_get_source_ip(
    hass: HomeAssistant, target_ip: str | UndefinedType = UNDEFINED
) -> str:
    """Get the source ip for a target ip."""
```

Adding this option puts an extra burden on the user having a bunch of IP addresses configured hidden inside config entry configurations.

balloob commented on homeassistant/components/voip/devices.py:
```diff
@@ -136,7 +136,11 @@ def async_get_or_create(self, call_info: CallInfo) -> VoIPDevice:
 
         dev_reg = dr.async_get(self.hass)
-        voip_id = call_info.caller_ip
+        voip_id = (
```
Can we just always do `uri` ? When would `caller_endpoint` not be available?

balloob commented on homeassistant/components/voip/devices.py:
```diff
@@ -136,7 +136,11 @@ def async_get_or_create(self, call_info: CallInfo) -> VoIPDevice:
+        voip_id = (
+            call_info.caller_endpoint.uri
+            if call_info.caller_endpoint
+            else call_info.caller_ip
```
Can we do an auto-migration here? So every incoming device, we check if it exists under the old ID, and if so, we migrate it to the new endpoint uri.

jaminh replied:
I tried to update it to do an auto-migration, but it would be difficult for me to test that since I already have new devices for all of my phones.

balloob reviewed (changes requested):
Can you break out the caller ID breaking change into its own PR? I also added a note how we can try to auto-migrate old devices automatically. Would be cool to add to that PR if possible.

balloob commented:
> Currently when announce is called HA will call the phone and go into the standard assist pipeline when the call is answered.

We should also make sure that we start with speaking the announcement message. Once we did that, Home Assistant should hang up. Users should use `start_conversation` if they want to continue talking.

jaminh committed:
Make VOIP host configuration optional

jaminh commented:
> Can you break out the caller ID breaking change into its own PR? I also added a note how we can try to auto-migrate old devices automatically. Would be cool to add to that PR if possible.

https://github.com/home-assistant/core/pull/135603

jaminh committed:
Add advanced options for Voip host and user

jaminh committed:
Remove create rtp server method

The create rtp server method is in voip_utils

jaminh committed:
Outgoing calls seem to be working

Made a queue of pipeline operations that can be added to in case
the announcement service is called during an ongoing call.

jaminh commented on homeassistant/components/voip/assist_satellite.py:
```diff
@@ -149,26 +173,110 @@ async def async_set_configuration(
+        _LOGGER.debug("Assist satellite connection made")
+        super().connection_made(transport)
+        if self._pipeline_task_queue.empty():
+            self._default_pipeline_task = asyncio.create_task(
```
Seems like there should be a better way to handle this but it seems to be working.

jaminh replied:
I think I solved this with put_nowait.

jaminh commented on homeassistant/components/voip/assist_satellite.py:
```diff
@@ -149,26 +173,110 @@ async def async_set_configuration(
+                )
+            except asyncio.QueueEmpty:
+                _LOGGER.debug("No task on queue")
+                self.disconnect()
```
I'll need to add something here to send a proper BYE message to end the call.

jaminh commented on homeassistant/components/assist_satellite/entity.py:
```diff
@@ -416,6 +416,7 @@ async def _resolve_announcement_media_id(
                 language=pipeline.tts_language,
                 options=tts_options,
             )
+            return AssistSatelliteAnnouncement(message, media_id, media_id_source)
```
Seems like this has the potential to affect other assist satellite components, but the VOIP announcements wouldn't work otherwise.

jaminh replied:
I added an internal_media_id property to the AssistSatelliteAnnouncement to hopefully mitigate this issue.

jaminh committed:
Improve task queue handling and add hang up

Improve task queue handling by using put_nowait for adding tasks to
avoid the need for creating a task for adding items. The task queue
should also be cleared when the call ends. If the call needs to end,
either because it is the end of an announcement or there was an
unrecoverable error, HA should initiate a hang up in an attempt to
cleanly end the call.

jaminh commented:
> > Currently when announce is called HA will call the phone and go into the standard assist pipeline when the call is answered.
> 
> We should also make sure that we start with speaking the announcement message. Once we did that, Home Assistant should hang up. Users should use `start_conversation` if they want to continue talking.

This PR is already pretty unwieldy as it is, I figure we can get this stable and add start_conversation support as a separate PR.

balloob commented on homeassistant/components/assist_satellite/entity.py:
```diff
@@ -96,7 +96,11 @@ class AssistSatelliteAnnouncement:
     media_id: str
     """Media ID to be played."""
 
+    internal_media_id: str
```
I've extracted this as https://github.com/home-assistant/core/pull/136141 and renamed to `original_media_id`

jaminh replied:
Should I rebase this branch to pick up that change or clean that up later?

balloob replied:
rebase.

balloob commented on homeassistant/components/voip/voip.py:
```diff
@@ -102,8 +105,19 @@ def __init__(self, hass: HomeAssistant, devices: VoIPDevices) -> None:
+    def on_call(self, call_info: CallInfo):
+        """Set up state when starting a call."""
+        device = self.devices.async_get_or_create(call_info)
+        if device is not None:
```
The response can never be `None`.

balloob commented on homeassistant/components/voip/voip.py:
```diff
@@ -102,8 +105,19 @@ def __init__(self, hass: HomeAssistant, devices: VoIPDevices) -> None:
+        """Set up state when starting a call."""
+        device = self.devices.async_get_or_create(call_info)
+        if device is not None:
+            device.current_call = call_info
```
Is this something that we want to pass to `set_active` instead? Having a call info then indicates it's active.

synesthesiam commented on homeassistant/components/wyoming/stt.py:
```diff
@@ -126,6 +126,9 @@ async def async_process_audio_stream(
                         text = transcript.text
                         break
 
+        except TimeoutError:
```
Why was this added?

jaminh replied:
Without it the call state wouldn't get properly cleaned up which I think was causing subsequent calls to not create a new RTP server and there would be no audio on the phone.

jaminh replied:
I removed it in the latest change, I am still having issues with it but it is not the same as before so that might have been a red herring when I added it the first time. I think I need to increase the timeout value because it seems like when I say a command it will make the tone like it heard the command, but then shortly after it hangs up on me from what I assume is caused by the timeout error.

jaminh replied:
I added this back in because hanging up after the regular command pipeline loop wouldn't work without it.

synesthesiam commented:
We need to add an implementation for this still, right?
`TypeError: Can't instantiate abstract class HassVoipDatagramProtocol without an implementation for abstract method 'on_hangup'`

jaminh commented:
> We need to add an implementation for this still, right? `TypeError: Can't instantiate abstract class HassVoipDatagramProtocol without an implementation for abstract method 'on_hangup'`

It should be implemented, line 134 homeassistant/components/voip/voip.py

jaminh committed:
Change set_is_active to take CallInfo

jaminh committed:
Remove host configuration

balloob commented:
It's not explicitely mentioned, but this PR requires https://github.com/home-assistant-libs/voip-utils/pull/25 to be merged and released right?

jaminh commented:
> It's not explicitely mentioned, but this PR requires [home-assistant-libs/voip-utils#25](https://github.com/home-assistant-libs/voip-utils/pull/25) to be merged and released right?

Yes, is there a way to specify an unreleased version in the manifest?

jaminh committed:
Use original media ID for announcement

jaminh committed:
Remove unnecessary audio streaming from announce

jaminh commented:
The pre-commit checks complain if I try to set the voip_utils requirement to point to my branch, so I left that as is.

I am still having an issue where when I hang up after doing the regular voice command pipeline stuff the TimeoutError isn't being caught in the _run_pipeline method which means the call isn't hung up. This means if I try to then play an announcement after that it thinks the previous call is ongoing and tries to play the announcement on that non-existent call. I still haven't figured out a way to get that to work such that it hangs up after I hang up my phone without it randomly hanging up in the middle of processing commands.

jaminh committed:
Restore catching/re-raising TimeoutError in STT

balloob commented:
Yeah we need to do a VoIP utils release for it to be mergeable.

balloob commented:
Do you have a stacktrace of where timeout is raised ?

jaminh commented:
> Do you have a stacktrace of where timeout is raised ?

Sorry I have been trying a lot of things to debug this issue and I'm having trouble reproducing it now and lost the stacktrace from my logs. From what I remember it seemed like the timeout was coming from the with timeout block in the stt_stream method/coroutine. There seem to be some kind of race conditions going on, I seem to recall there being instances of CancelledError being in the stack trace as well. It seems like the way it is now if I hang up the phone depending on when the TimeoutError is thrown from the stt_stream function it doesn't get caught in the _run_pipeline method in the voip assist_satellite class, which means the call doesn't get hung up on the HA side.

I have a suspicion that there is something that is sometimes cancelling the parent task of the running stt_stream instance, and if that happens while that is within the timeout block it is translating that into a TimeoutError. If that is the case that would explain why I would sometimes have the call hangup on me in the middle of processing. In that case I think we could solve the issue by finding a way to differentiate between the CancelledError and an actual TimeoutError, and only hangup in the case of an actual Timeout. Also if the stt_stream isn't running for the entire duration of the call then if the phone hangs up outside of the time stt_stream is running there isn't a way for HA to detect that the call was ended, which is another situation I think I have experienced.

Long story short it seems like there are still issues around detecting when a phone hangs up and I'm still trying to wrap my head around asyncio.

jaminh commented:
Investigating the issue more, I added an except clause in the _run_pipeline method and while trying to send a command I heard the tone after my command that would indicate that it heard the command successfully, but it then hung up immediately after. It logged out the following exception from the except clause I added:

2025-01-25 14:47:41.992 ERROR (MainThread) [homeassistant.components.voip.assist_satellite] Unexpected error during pipeline
Traceback (most recent call last):
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/components/voip/assist_satellite.py", line 284, in _run_pipeline
    await self.async_accept_pipeline_from_satellite(
        audio_stream=stt_stream(),
    )
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/components/assist_satellite/entity.py", line 310, in async_accept_pipeline_from_satellite
    await self._pipeline_task
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/components/assist_pipeline/__init__.py", line 135, in async_pipeline_from_audio_stream
    await pipeline_input.execute()
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/components/assist_pipeline/pipeline.py", line 1482, in execute
    tts_input = await self.run.recognize_intent(
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<3 lines>...
    )
    ^
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/components/assist_pipeline/pipeline.py", line 1084, in recognize_intent
    conversation_result = await conversation.async_converse(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<7 lines>...
    )
    ^
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/components/conversation/agent_manager.py", line 110, in async_converse
    result = await method(conversation_input)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/components/conversation/entity.py", line 47, in internal_async_process
    return await self.async_process(user_input)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/components/conversation/default_agent.py", line 367, in async_process
    intent_result = await self.async_recognize_intent(user_input)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/components/conversation/default_agent.py", line 330, in async_recognize_intent
    result = await self.hass.async_add_executor_job(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<7 lines>...
    )
    ^
asyncio.exceptions.CancelledError

So it seems like something is cancelling the task, but I haven't been able to see what might be doing it and why.

jaminh committed:
Retry STT event reads if task cancelled

jaminh commented:
I was able to increase the reliability of the pipeline by retrying the event read operation if the task is cancelled in the wyoming stt class. I still occasionally get CancelledErrors in other locations though, for example:

2025-01-25 21:28:30.039 ERROR (MainThread) [homeassistant.components.voip.assist_satellite] Unexpected error during pipeline
Traceback (most recent call last):
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/components/voip/assist_satellite.py", line 284, in _run_pipeline
    await self.async_accept_pipeline_from_satellite(
        audio_stream=stt_stream(),
    )
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/components/assist_satellite/entity.py", line 310, in async_accept_pipeline_from_satellite
    await self._pipeline_task
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/components/assist_pipeline/__init__.py", line 135, in async_pipeline_from_audio_stream
    await pipeline_input.execute()
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/components/assist_pipeline/pipeline.py", line 1482, in execute
    tts_input = await self.run.recognize_intent(
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<3 lines>...
    )
    ^
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/components/assist_pipeline/pipeline.py", line 1084, in recognize_intent
    conversation_result = await conversation.async_converse(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<7 lines>...
    )
    ^
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/components/conversation/agent_manager.py", line 110, in async_converse
    result = await method(conversation_input)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/components/conversation/entity.py", line 47, in internal_async_process
    return await self.async_process(user_input)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/components/conversation/default_agent.py", line 368, in async_process
    return await self._async_process_intent_result(intent_result, user_input)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/components/conversation/default_agent.py", line 441, in _async_process_intent_result
    intent_response = await intent.async_handle(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<10 lines>...
    )
    ^
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/helpers/intent.py", line 144, in async_handle
    result = await handler.async_handle(intent)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/helpers/intent.py", line 1003, in async_handle
    response = await self.async_handle_states(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        intent_obj, match_result, match_constraints, match_preferences
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/srv/homeassistant/lib64/python3.13/site-packages/homeassistant/helpers/intent.py", line 1068, in async_handle_states
    await service_coro
  File "/usr/lib64/python3.13/asyncio/tasks.py", line 630, in _wait_for_one
    f = await self._done.get()
        ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.13/asyncio/queues.py", line 186, in get
    await getter
asyncio.exceptions.CancelledError

I initially thought it had to do with the stt_stream timeout but I think that is likely a red herring based on putting try/excepts in that function for CancelledError. Seems like there is some kind of race condition going on and when I was changing the stt timeout value it may have changed at what point in the process the cancel was happening which made it appeared to fix the issue but it wasn't really having the direct impact I thought it was. Maybe we just need to make pretty much the whole assist pipeline more tolerant of task cancellation?

balloob commented:
Yeah, we should always be able to expect task cancellation. It's how asyncio works: any task can be cancelled by the code that spawned it. We are doing this actively inside `assist_satellite` entity when the phone calls us, or a new announce is sent to the phone: https://github.com/home-assistant/core/blob/dev/homeassistant/components/assist_satellite/entity.py#L314

balloob commented:
`TimeoutError` is raised when we set a max execution time, and that time is exceeded. `CancelledError` is raised when someone calls `cancel` on the task itself, or one of the parent tasks that is waiting for this code. 

When either happens, the exception is raised wherevever the task is currently awaiting. Fun fact: you can catch `CancelledError` and ignore the cancellation. You can either do that manually by catching it and going on with your business. Or you wrap the code you don't want to be cancelled in an [`asyncio.shield`](https://docs.python.org/3/library/asyncio-task.html#shielding-from-cancellation).

jaminh committed:
Catch CancelledError in pipeline

If CancelledError is caught, wait a little bit longer to see if the
pipeline task finishes, then restart the pipeline.

jaminh commented:
OK, I updated it to catch CancelledError, wait a little bit longer to finish the accept pipeline task, and then restart the pipeline. After making that change I haven't actually been able to reproduce the CancelledError, but it is running more reliably so in that sense mission accomplished I guess? I'll continue testing on my end, it would be nice to get some feedback from other people's testing to make sure I didn't break things for everyone else though.

synesthesiam commented:
Please rebase after this is merged: https://github.com/home-assistant/core/pull/136648

synesthesiam committed:
Merge branch 'dev' into voip-assist-sat-announce

synesthesiam commented:
@jaminh I'll take the PR from here over the finish line. Thanks for all of your efforts :slightly_smiling_face:

balloob commented:
Closing this in favor of https://github.com/home-assistant/core/pull/136781. Announce is live in the next nightly being build now 🎉
